sdks/python/src/opik/evaluation/metrics/conversation/conversation_thread_metric.py

```diff
@@ -0,0 +1,24 @@
+from typing import Union, List, Any, Dict
+
+from .. import base_metric, score_result
+
+
+class ConversationThreadMetric(base_metric.BaseMetric):
+    """Abstract base class for all conversation thread metrics."""
+
+    def score(
+        self, conversation: List[Dict[str, str]], **kwargs: Any
```

Comment:
Let's make the type a bit more verbose and go with `List[Dict[Literal["role", "content"], str]]`
Still not the most accurate one, but quite explanatory for those who finds these key words familiar.